org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XSISchemaModel.java

```diff
@@ -0,0 +1,89 @@
+/**
+ *  Copyright (c) 2018 Red Hat, Inc. and others.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  Contributors:
+ *  Red Hat Inc. - initial API and implementation
+ */
+package org.eclipse.lsp4xml.services;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionItemKind;
+import org.eclipse.lsp4j.InsertTextFormat;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4xml.dom.Element;
+import org.eclipse.lsp4xml.dom.XMLDocument;
+import org.eclipse.lsp4xml.services.extensions.ICompletionRequest;
+import org.eclipse.lsp4xml.services.extensions.ICompletionResponse;
+
+/**
+ * This class holds values that represent the XSI xsd.
+ * Can be seen at https://www.w3.org/2001/XMLSchema-instance
+ */
+public class XSISchemaModel {
+	
+	public static void computeCompletionResponses(ICompletionRequest request, 
+			ICompletionResponse response, Range editRange, XMLDocument document) {
+		String snippet = "";
+		boolean isSnippetsSupported = request.getCompletionSettings().isCompletionSnippetsSupported();
+		if(isSnippetsSupported) {
+			snippet = "$0";
+		}
+		String actualPrefix = document.getSchemaInstancePrefix();
+		String name;
+		String documentation;
+		Element root = document.getDocumentElement();
+		boolean schemaLocationExists = document.hasSchemaLocation();
+		boolean noNamespaceSchemaLocationExists = document.hasNoNamespaceSchemaLocation();
+		//Indicates that no values are allowed inside an XML element
+		if(!attributeAlreadyExists(root, actualPrefix, "nil")) {
+			documentation = "Indicates that no values are allowed inside an XML element";
```

Comment:
"Indicates that no values are allowed inside an XML element. Valid values are `true`/`false`"